Content.Server/Electrocution/ElectrocutionSystem.cs

```diff
@@ -186,7 +187,18 @@ private void OnLightAttacked(EntityUid uid, PoweredLightComponent component, Att
         if (_meleeWeapon.GetDamage(args.Used, args.User).Empty)
             return;
 
-        DoCommonElectrocution(args.User, uid, component.UnarmedHitShock, component.UnarmedHitStun, false);
+        var siemens = 1f;
+
+        // Check for insualted component on the entity triggering the action or on the gloves
+        if (TryComp<InsulatedComponent>(args.User, out var insulation) ||
+        (TryComp<InventoryComponent>(args.User, out var inventory) &&
+         _inventory.TryGetSlotEntity(args.User, "gloves", out var gloves, inventory) &&
+         TryComp<InsulatedComponent>(gloves, out insulation)))
+        {
+            siemens = insulation.Coefficient;
+        }
+
+        DoCommonElectrocution(args.User, uid, component.UnarmedHitShock, component.UnarmedHitStun, false, siemens);
```

Comment:
Spelling dude.

Also if there isn't there needs to be a helper method for this.